modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/metadata/IgniteMdCollation.java

```diff
@@ -342,7 +343,8 @@ private IgniteMdCollation() {
             fieldCollations.clear();
             boolean skip = false;
             for (RelFieldCollation ifc : ic.getFieldCollations()) {
-                final Collection<Integer> integers = targets.get(ifc.getFieldIndex());
+                final Collection<Integer> integers = targets.getOrDefault(ifc.getFieldIndex(),
+                        Collections.emptyList());
```

Comment:
'targets' actually contains IntArrayList, isn't it?
```suggestion
                final IntArrayList integers = targets.getOrDefault(ifc.getFieldIndex(),
                        IntLists.emptyList());
```

Reply:
fieldCollations.add(ifc.withFieldIndex(integers.iterator().**nextInt**()));